src/player/mod.rs

```diff
@@ -0,0 +1,65 @@
+#[cfg(test)]
+mod tests;
+
+use crate::state::WINDOW_HEIGHT;
+
+pub const PLAYER_SPEED: f32 = 6.5;
+pub const PADDLE_WIDTH : f32 = 30.0;
+pub const PADDLE_HEIGHT: f32 = 150.0;
+
+#[derive(Default)]
+pub struct Player {
+    pub position: (f32,f32),
+    moving: bool,
+    velocity: f32,
+}
+
+impl Player {
+    pub fn new(x: f32, y: f32) -> Self {
+        Self {
+            position: (x, y),
+            moving: false,
+            velocity: 0.0,
+        }
+    }
+
+    pub fn move_down(&mut self) {
+        self.moving = true;
+        self.velocity = PLAYER_SPEED;
+    }
+
+    pub fn move_up(&mut self) {
+        self.moving = true;
+        self.velocity = -PLAYER_SPEED;
+    }
+
+    pub fn stop_moving(&mut self) {
+        self.moving = false;
+    }
+
+    pub fn is_moving(&self) -> bool {
+        self.moving
```

Comment:
if the field is named  `is_moving` we dont need a function for these

Reply:
Fixed!